src/main/java/com/spotify/docker/Utils.java

```diff
@@ -65,20 +65,34 @@
   }
 
   public static void pushImage(DockerClient docker, String imageName, Log log,
-                               final DockerBuildInformation buildInfo)
-      throws MojoExecutionException, DockerException, IOException, InterruptedException {
-      log.info("Pushing " + imageName);
-
-    final AnsiProgressHandler ansiProgressHandler = new AnsiProgressHandler();
-    final DigestExtractingProgressHandler handler = new DigestExtractingProgressHandler(
-        ansiProgressHandler);
-
-    docker.push(imageName, handler);
-
-    if (buildInfo != null) {
-      final String imageNameWithoutTag = parseImageName(imageName)[0];
-      buildInfo.setDigest(imageNameWithoutTag + "@" + handler.digest());
-    }
+                               final DockerBuildInformation buildInfo,
+                               int retryCount, int retryTimeout)
+          throws MojoExecutionException, DockerException, IOException, InterruptedException {
+    log.info("Pushing " + imageName);
+    int attempt = 0;
+    do {
+      final AnsiProgressHandler ansiProgressHandler = new AnsiProgressHandler();
+      final DigestExtractingProgressHandler handler = new DigestExtractingProgressHandler(
+              ansiProgressHandler);
+
+      try {
+        docker.push(imageName, handler);
+      } catch (DockerException e) {
```

Comment:
It bubbles up as a DockerException. I have added a comment about that in the code.